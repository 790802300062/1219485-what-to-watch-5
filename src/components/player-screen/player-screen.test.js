import React from "react";
import renderer from "react-test-renderer";
import {filmListMock, noop} from "../../test-data";
import PlayerScreen from "./player-screen";

describe(`PlayerScreen render`, () => {
  it(`renders PlayerScreen component with the 1st movie`, () => {
    const tree = renderer
      .create(
          <PlayerScreen
            film={filmListMock[0]}
            onExitButtonClick={noop}
          />,
          {
            createNodeMock: () => {
              return {};
            }
          }
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`renders PlayerScreen component with the 3rd movie`, () => {
    const tree = renderer
      .create(
          <PlayerScreen
            film={filmListMock[2]}
            onExitButtonClick={noop}
          />,
          {
            createNodeMock: () => {
              return {};
            }
          }
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
