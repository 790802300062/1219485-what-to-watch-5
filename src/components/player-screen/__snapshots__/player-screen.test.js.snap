// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayerScreen render renders PlayerScreen component with the 1st movie 1`] = `
<div
  className="player"
>
  <video
    className="player__video"
    muted={false}
    poster="https://assets.htmlacademy.ru/intensives/javascript-3/film/background/bronson.jpg"
    src="http://media.xiph.org/mango/tears_of_steel_1080p.webm"
  />
  <button
    className="player__exit"
    onClick={[Function]}
    type="button"
  >
    Exit
  </button>
  <div
    className="player__controls"
  >
    <div
      className="player__controls-row"
    >
      <div
        className="player__time"
      >
        <progress
          className="player__progress"
          max="100"
          value={0}
        />
        <div
          className="player__toggler"
          style={
            Object {
              "left": "0%",
            }
          }
        >
          Toggler
        </div>
      </div>
      <div
        className="player__time-value"
      >
        20:00:00
      </div>
    </div>
    <div
      className="player__controls-row"
    >
      <button
        className="player__play"
        onClick={[Function]}
        type="button"
      >
        <svg
          height="19"
          viewBox="0 0 19 19"
          width="19"
        >
          <use
            xlinkHref="#play-s"
          />
        </svg>
        <span>
          Play
        </span>
      </button>
      <div
        className="player__name"
      >
        Bronson
      </div>
      <button
        className="player__full-screen"
        onClick={[Function]}
        type="button"
      >
        <svg
          height="27"
          viewBox="0 0 27 27"
          width="27"
        >
          <use
            xlinkHref="#full-screen"
          />
        </svg>
        <span>
          Full screen
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`PlayerScreen render renders PlayerScreen component with the 3rd movie 1`] = `
<div
  className="player"
>
  <video
    className="player__video"
    muted={false}
    poster="https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Midnight_Special.jpg"
    src="http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4"
  />
  <button
    className="player__exit"
    onClick={[Function]}
    type="button"
  >
    Exit
  </button>
  <div
    className="player__controls"
  >
    <div
      className="player__controls-row"
    >
      <div
        className="player__time"
      >
        <progress
          className="player__progress"
          max="100"
          value={0}
        />
        <div
          className="player__toggler"
          style={
            Object {
              "left": "0%",
            }
          }
        >
          Toggler
        </div>
      </div>
      <div
        className="player__time-value"
      >
        16:00:00
      </div>
    </div>
    <div
      className="player__controls-row"
    >
      <button
        className="player__play"
        onClick={[Function]}
        type="button"
      >
        <svg
          height="19"
          viewBox="0 0 19 19"
          width="19"
        >
          <use
            xlinkHref="#play-s"
          />
        </svg>
        <span>
          Play
        </span>
      </button>
      <div
        className="player__name"
      >
        Midnight Special
      </div>
      <button
        className="player__full-screen"
        onClick={[Function]}
        type="button"
      >
        <svg
          height="27"
          viewBox="0 0 27 27"
          width="27"
        >
          <use
            xlinkHref="#full-screen"
          />
        </svg>
        <span>
          Full screen
        </span>
      </button>
    </div>
  </div>
</div>
`;
